### PR TITLE
Introduce a new pre-install script

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Add a `pre-install` script that is install/linking git hooks in a non-destructive way.

### CODE REVIEW

**Comment by wavesoft** [comment]
LGTM! ✨ 💯

File: scripts/utils/git [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# Install/link GIT hooks
+# Example: install_hooks "pre-commit pre-push"
+install_hooks() {
+  # Check if .git/hooks directory exists
+  if [ ! -d ".git/hooks" ]
+  then
+    warning "Couldn't install hooks as the directory ${NC}.git/hooks/${RED}"\
+              "doesn't exists"
+    return
```
Comment (nLight) [inline scripts/utils/git]: We could also create`hooks` directory if `.git` exists

**Comment by nLight** [comment]
Please consider my comment and merge it, good job!

File: scripts/utils/git (hunk 1 above)
Comment (orlandohohmeier) [inline scripts/utils/git]: As far as I know there is by default always a `.git/hooks` directory, thus I decided not to add the needed extra command to create that directory if it doesn't exists. Do you think we should add this case?

Comment (nLight) [inline scripts/utils/git]: I'm fine wth that assumption

**Comment by nLight** [comment]
Looks like I can't delete the branch @orlandohohmeier

**Comment by nLight** [comment]
Ok, it became possible and I did it. Interesting...

**Pull request merged**